### DISCUSSION TITLE
Is there a published API for UTM so I can interact with it via API calls?

### DISCUSSION THREAD

**Discussion by EdgeSync** [utmapp/UTM#5317] [General]
Hey there, 

I was curios to know, is there any API documentation on how to interact with UTM via scripts/programs?

Basically I would like to be able to communicate with the hypervisor layer of UTM through C or Python. To begin i'd like to be able to query UTM and see what VM's are running, start/stop vm's, and then progressing to things like automatically deploying VM's, and progressing further to actually being able to talk to VM through UTM and have it return information about what is happening inside the VM (maybe a list of processes, ram usage, that kind of thing, give me some information on the kernel, and so on).

I'm just doing some research on automating some security for hypervisors, and though this might be a good place to ask. I have lots of experience working with web api's, but system level calls are something I'm just getting into. Hopefully I've explained my ask adequately :)

If anyone had any ideas or could point me where I should be looking, that'd be great!

Thanks!

**Comment by osy** [comment]
https://docs.getutm.app/scripting/scripting/

**Reply by EdgeSync** [reply]
you are a champ! thank you very much!
